### PR TITLE
Break metric, angular, through-vision and isovist analyses out of pointmap

### PR DESCRIPTION
**Pull request by orange-vertex**
~~Isovist analysis is carried out using the currently visible drawing layers, hence for this to work the interface had to be extended to also accept the MetaGraph. The other solutions I can think of is to make a separate interface for modules that want to use the MetaGraph, or for the isovist analysis to use the drawing layers attached to the pointmap itself~~

Changed to make the isovist analysis take the drawing layers from the pointmap itself

### CODE REVIEW

File: salalib/vgamodules/vgavisualglobal.h [hunk 1]
```diff
@@ -19,27 +19,24 @@
...
-    std::string getAnalysisName() const override {
-        return "Global Visibility Analysis";
-    }
+class VGAVisualGlobal : IVGA {
+  public:
+    std::string getAnalysisName() const override { return "Global Visibility Analysis"; }
     bool run(Communicator *comm, const Options &options, PointMap &map, bool simple_version) override;
-    void extractUnseen(Node& node, PixelRefVector& pixels, depthmapX::RowMatrix<int>& miscs,
-                       depthmapX::RowMatrix<PixelRef>& extents) {
+    void extractUnseen(Node &node, PixelRefVector &pixels, depthmapX::RowMatrix<int> &miscs,
```
Comment (blsemo) [inline salalib/vgamodules/vgavisualglobal.h]: If you touch this file - I don't think a chunky function like this should be in the header...

File: salalib/ivga.h [hunk 2]
```diff
@@ -17,17 +17,17 @@
 
 // Interface to handle different kinds of VGA analysis
 
-#include "salalib/pointdata.h"
+#include "salalib/mgraph.h"
 #include "salalib/options.h"
+#include "salalib/pointdata.h"
```
Comment (blsemo) [inline salalib/ivga.h]: Too many blank lines

File: salalib/ivga.h [hunk 3]
```diff
@@ -17,17 +17,17 @@
...
 #include "genlib/comm.h"
 
 #include <string>
 
-class IVGA
-{
-public:
+class IVGA {
+  public:
     virtual std::string getAnalysisName() const = 0;
```
Comment (blsemo) [inline salalib/ivga.h]: This should probably return a `string const &` rather than a copy.

Comment (orange-vertex) [reply salalib/ivga.h]: This is used like this:
`std::string getAnalysisName() const override { return "Local Visibility Analysis"; }`
so there's no object to return a reference to..

**Pull request merged**